Userland/Libraries/LibPthread/semaphore.cpp

```diff
@@ -47,9 +47,20 @@ int sem_destroy(sem_t* sem)
     return 0;
 }
 
-int sem_getvalue(sem_t*, int*)
+int sem_getvalue(sem_t* sem, int* sval)
 {
-    VERIFY_NOT_REACHED();
+    errno = pthread_mutex_trylock(&sem->mtx);
+    if (errno == EBUSY) {
+        *sval = 0;
+        return 0;
+    } else if (errno != 0)
```

Comment:
No need for `else` after `return`.

Reply:
In theory `pthread_mutex_trylock()` could return something else other than `EBUSY` and `0` though, right? I know our `pthread_mutex_trylock()` doesn't currently, but maybe later on.

Reply:
I just meant that you can write this as
```c++
if (errno == EBUSY) {
    // EBUSY things
}
if (errno != 0)
    // not EBUSY things
```

Reply:
Oh right, good point. I'll change that. :+1:

Reply:
@alimpfard Does this get optimized as expected? Meaning is it going to essentially do two `cmp` if the first condition is true? Just wondering because `errno` is a special (tls?) variable. I usually use `else if` for those reasons and only do two `if` if I'm e.g. `return`ing in the first condition.

Reply:
I just disassembled `sem_getvalue` for the fun of it and it does appear to get optimized properly. The compiler stores the result from the call into the TLS variable (which arguably isn't necessary if the result is 0 - but I'm not sure if that's worth optimizing) but keeps using the value it has in a register for the two CMPs, i.e. it does _not_ re-load the value from the TLS segment.

Reply:
Well, I decided that I do care after all, so I added a temporary variable instead of just storing the result right into `errno`.